test/models/survey_test.exs

```diff
@@ -111,4 +117,419 @@ defmodule Ask.SurveyTest do
 
     assert survey_channels_ids == [channel_1.id, channel_2.id, channel_3.id]
   end
+
+  describe "retries histogram" do
+    test "flow with no retries" do
+      mode = ["sms"]
+      survey = insert(:survey, %{fallback_delay: "10h"})
+      stats = %{survey_id: survey.id} |> RetryStat.stats()
+      %{flow: flow} = survey |> Survey.retries_histogram(stats, [mode], Timex.now)
+
+      assert flow == [%{type: mode, delay: 0}, %{type: "end", delay: 10, label: "10h"}]
+    end
+
+    defp flow_with_retries("ivr" = mode) do
+      survey = insert(:survey, Map.put(retry_configuration(mode, "1h 2h 3h"), :fallback_delay, "5h"))
+      stats = %{survey_id: survey.id} |> RetryStat.stats()
+      %{flow: flow} = survey |> Survey.retries_histogram(stats, [mode], Timex.now)
+
+      assert flow == [
+               %{type: mode, delay: 0},
+               %{type: mode, delay: 1, label: "1h"},
+               %{type: mode, delay: 2, label: "2h"},
+               %{type: mode, delay: 3, label: "3h"}
+             ]
+    end
+
+    defp flow_with_retries(mode) do
+      survey = insert(:survey, Map.put(retry_configuration(mode, "1h 2h 3h"), :fallback_delay, "5h"))
+      stats = %{survey_id: survey.id} |> RetryStat.stats()
+      %{flow: flow} = survey |> Survey.retries_histogram(stats, mode, Timex.now)
+
+      assert flow == [
+               %{type: mode, delay: 0},
+               %{type: mode, delay: 1, label: "1h"},
+               %{type: mode, delay: 2, label: "2h"},
+               %{type: mode, delay: 3, label: "3h"},
+               %{type: "end", delay: 5, label: "5h"}
+             ]
+    end
+
+    test "flow with retries" do
+      flow_with_retries("ivr")
+    end
```

Comment:
`flow_with_retries` is only being used once, and the second definition isn't used at all. There's probably something missing here 😇